src/util/KeyHelper.ts

```diff
@@ -0,0 +1,76 @@
+/* tslint:disable:no-relative-imports */
+import { EncodeBuffer } from 'base-x'
+import * as bs58 from 'bs58'
+import { pki, random } from 'node-forge'
+
+import { SigningAlgorithm } from '../Interfaces'
+
+interface KeyPair {
+  readonly publicKey: string
+  readonly privateKey: string
+}
+
+interface PrivateKey {
+  readonly n: number
+  readonly e: number
+}
+
+const generateRsaPublicKeyFromPrivateKey = ({ privateKey }: { privateKey: PrivateKey }) =>
+  pki.rsa.setPublicKey(privateKey.n, privateKey.e)
+
+const Algorithms: any = {
```

Comment:
Not sure about this, but maybe if you remove the `any` TS can infer a type for this

Reply:
engine is an object that has the generateKeyPair function

Reply:
And Rsa.generateKeyPair takes a completely different set of arguments from Ed25519.generateKeyPair